### PR TITLE
subject+task

### PR DESCRIPTION
**Pull request by PhamNgocManh**
subject + task

### CODE REVIEW

File: app/controllers/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def destroy
+    Subject.find(params[:id]).destroy
+    flash[:success] = "Subject deleted"
+    redirect_to subjects_url
+  end
+
+  private
+    def subject_params
```
Comment (dieunb) [inline app/controllers/subjects_controller.rb]: indent function `subject_params`

File: app/models/subject.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Subject < ActiveRecord::Base
+  has_many :tasks, dependent: :destroy
+  validates :name, presence: true, length: {maximum: 50}, uniqueness:true
+  accepts_nested_attributes_for :tasks, reject_if: proc { |attributes| attributes['name'].blank? }, :allow_destroy => true
```
Comment (dieunb) [inline app/models/subject.rb]: length

File: app/models/task.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Task < ActiveRecord::Base
+  belongs_to :subject
+  validates :name, presence: true, length: { maximum: 100 }
```
Comment (dieunb) [inline app/models/task.rb]: remove space before `}` and after `{`

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -15,3 +15,14 @@
 //= require turbolinks
 //= require_tree .
 //= require bootstrap
+function remove_fields(field) {
+  $(field).prev().val("true");
+  $(field).parent().hide()
+}
+function add_fields(link, name) {
```
Comment (dieunb) [inline app/assets/javascripts/application.js]: add a new line above this function

File: app/helpers/subjects_helper.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+module SubjectsHelper
+  def creat_add_task_button(f, tasks)
+    new_object = f.object.class.reflect_on_association(tasks).klass.new
+    fields = f.fields_for(tasks, new_object) do |builder|
+      render "task_fields", :f => builder
```
Comment (dieunb) [inline app/helpers/subjects_helper.rb]: `f: builder`

File: app/models/subject.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class Subject < ActiveRecord::Base
+  has_many :tasks, dependent: :destroy
+  validates :name, presence: true, length: {maximum: 50}, uniqueness:true
+  accepts_nested_attributes_for :tasks,
+    reject_if: proc { |attributes| attributes['name'].blank? },
```
Comment (dieunb) [inline app/models/subject.rb]: remove space before `}` and after `{`

File: app/views/subjects/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    <%= form_for @subject do |f| %>
+      <%= render "shared/error_messages", object: f.object %>
+      <p>
+      <%= f.label :name %>
+      <%= f.text_field :name, class: "form-control" %>
+      <%= f.label :instruction %>
+      <%= f.text_area :instruction, cols: 20, rows: 10, class: "form-control" %>
+      </p>
+      <%= f.fields_for :tasks do |task| %>
+        <%= render "task_fields", :f => task %>
```
Comment (dieunb) [inline app/views/subjects/new.html.erb]: `f: task`

File: app/models/subject.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+class Subject < ActiveRecord::Base
+  has_many :tasks, dependent: :destroy
+  validates :name, presence: true, length: {maximum: 50}, uniqueness:true
+  accepts_nested_attributes_for :tasks,
+    reject_if: proc {|attributes| attributes['name'].blank?},
+    :allow_destroy => true
```
Comment (chuanhtuan) [inline app/models/subject.rb]: allow_destroy: true

**Pull request merged**